### DISCUSSION TITLE
Detected Multiple Instance of Livewire Running

### DISCUSSION THREAD

**Discussion by menarasb** [livewire/livewire#6211] [1. Help] [answered]
Detected Multiple Instance of Livewire Running when i use return $this->redirect('/', navigate:true)

**Comment by PhiloNL** [comment]
Chances are that you are loading Alpine/Livewire manually. Can you replicate the problem with a fresh installation of Laravel + the latest beta version?

**Reply by menarasb** [reply]
i don't use it manually, wire:navigate doing well when using <a> tag, but when i use redirect('',navigate:true) in a component function it becomes a problem

**Comment by joshhanley** [comment]
@menarasb any chance you could provide an example repo that demonstrates the issue so we can investigate? Thanks!

**Reply by menarasb** [reply]
```php
public function login()
    {
        if (Auth::attempt(['username' => $this->username, 'password' => $this->password])) {
            session()->regenerate();
            return $this->redirect('/dashboard', navigate:true);
        } else 
        { 
            dd('fail');
        }
    }
``` 

it happens when i use Laravel Auth

**Comment by WendellAdriel** [comment]
The same is happening to me.
I don't have Alpine or Livewire inject manually anywhere in the application, but when redirecting with `$this->redirect` or `$this->redirectRoute` it's doing the same.

**Reply by joshhanley** [reply]
@WendellAdriel any chance you could share a demo repo so I can have a look?

**Reply by WendellAdriel** [reply]
@joshhanley I'll check if I have some time to create a demo repo because the one I'm having this issue is a private repo that I can't share.

**Comment by WendellAdriel** [comment]
@joshhanley I think this and #6107 are the same so I'll post in both just in case.
I tried to recreate the issue in a demo app, but weirdly I can't recreate the behaviour on it.

Here's the repo if you or @menarasb want to use it as a demo app:
https://github.com/WendellAdriel/livewire-navigate-issue

**Reply by WendellAdriel** [reply]
For sure that I removed a lot of things, but this is kinda what I have in my project that's not working.

Two layouts, app and dashboard with almost the same content as the ones I have in the demo app, just some specific stuff and styling with TailwindCSS.

Tried to create it as simple as possible but also kinda trying to follow what I have in the project I'm getting this issue

**Reply by thinkverse** [reply]
I submitted a PR to the reproduction and added the session regeneration, now the Livewire team should have a workable reproduction to test with if they need it.

**Reply by WendellAdriel** [reply]
Amazing, thanks man!
Already merged it!

**Comment by wiyatnanto** [comment]
The same is happening to me, buat after some tries it turns out that when the session is recreated it makes it duplicate livewire and alpinejs

**Reply by WendellAdriel** [reply]
Oh, now that you say that I saw this, it's happening when the session is recreated/regenerated

**Reply by menarasb** [reply]
yeahh that's right

**Comment by augustineasante88** [comment]
In livewire config I set injects assets to false and I manually included @livewirescripts and @livewirestyles in the app.blade and the issue is gone

**Reply by krismanning** [reply]
This was the solution for my problem also

**Comment by ZermattChris** [comment]
I'm new to both Laravel and Livewire...

Had been working happily away, when I noticed this dreaded "Loaded Livewire & Alpine twice" error in the browser's console.

I've gone through every online comment and suggestion, but still couldn't get the errors to stop...
I have a standard install of Jetstream + Livewire. 
After lots of random attempts (not really knowing what/where to change), I've ended up with this in my app.js file:

```
/**
 * I was getting Livewire & Alpine loaded 2x error messages in the browser console.
 * Have commented out the import statements below... Errors gone for the moment.
 */

//import { createApp } from "vue/dist/vue.esm-bundler.js";
//import {Alpine, Livewire} from '../../vendor/livewire/livewire/dist/livewire.esm';

//console.log(Livewire)
//console.log(Alpine)

// createApp()
//     .use()
//     .mount("#app");

//Livewire.start()
```

Not really sure where the Vue install came from, but both Alpine & Livewire were already available when this script is run.
If my beginner logic is correct, then it's the Livewire setting:

```
'inject_assets' => true,
```
that's injecting Alpine & Livewire into the system. Having the app.js file do it again is causing the issues. 

Fingers crossed :-)

**Comment by discoverlance-com** [comment]
I am experiencing this issue. I have updated my livewire config to stop the assets injection. It works when I don't pass any nonce to the script. But immediately I pass a nonce to the livewireScript for my csp, I get the multiple instances of livewire running error in my console.

```blade
{{--  With nonce, error reappears --}}
 @livewireScripts(['nonce' => csp_nonce()])

```

```blade
{{--  Without nonce, no errors show --}}
 @livewireScripts()

```

**Reply by joshhanley** [reply]
@discoverlance-com your issue was recently fixed in PR #8793 which will be available in the next release.

**Comment by joshhanley** [comment, marked as answer]
We're in the process of investigating all wire:navigate issues and I have tested out your scenario. I believe this has been fixed so I'm going to close this. If you're still experiencing problems, please open a new discussion. Thanks!
